### PR TITLE
Allow testing of velum from git checkout

### PR DESCRIPTION
**Pull request by flavio**
This PR makes possible to run a velum instance based on a git checkout. It also allows to do local development against velum in a similar fashion of the dev-env.

 Quite some parts of the PR are taken straight from the dev-env repository.

### CODE REVIEW

File: caasp-kvm/tools/download_image.py [hunk 1]
```diff
@@ -39,24 +46,31 @@ def get_filename(url):
...
     url_base = {
-        'release': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10/images/',
-        'staging_a': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/A/images/',
-        'staging_b': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/B/images/',
-        'devel': 'http://download.suse.de/ibs/Devel:/CASP:/1.0:/ControllerNode/images/',
+        'release': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10/',
+        'staging_a': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/A/',
+        'staging_b': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/B/',
+        'devel': 'http://download.suse.de/ibs/Devel:/CASP:/1.0:/ControllerNode/',
+        'head': 'http://download.suse.de/ibs/Devel:/CASP:/Head:/ControllerNode/',
```
Comment (kiall) [inline caasp-kvm/tools/download_image.py]: This is out of date + needs rebase - we've already dropped in the head repo (using "devel" as the name, as it's used in all the different tools etc)

File: caasp-kvm/tools/build-velum-image.sh [hunk 2]
```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
```
Comment (kiall) [inline caasp-kvm/tools/build-velum-image.sh]: can we add a `set -euo pipefail` line here? The old tooling suffered from continuing in spite of failures, and using this at the top of every bash script (and fixing any issues it uncovers) tends to help prevent these missed failures

File: caasp-kvm/tools/download_image.py [hunk 3]
```diff
@@ -39,28 +46,35 @@ def get_filename(url):
...
     url_base = {
-        'release': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP3:/Update:/Products:/CASP20/images/',
+        'release': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10/',
         # Staging images do not yet exist for 2.0
-        # 'staging_a': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/A/images/',
-        # 'staging_b': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/B/images/',
-        'devel': 'http://download.suse.de/ibs/Devel:/CASP:/Head:/ControllerNode/images/',
+        #'staging_a': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/A/',
+        #'staging_b': 'http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10:/Staging:/B/',
+        'devel': 'http://download.suse.de/ibs/Devel:/CASP:/1.0:/ControllerNode/',
```
Comment (kiall) [inline caasp-kvm/tools/download_image.py]: In all the other tools devel==the head repo now.. Can we leave as is for the moment, and we'll update everything at once?

File: caasp-kvm/cluster.tf [hunk 4]
```diff
@@ -182,6 +238,8 @@ resource "libvirt_volume" "master" {
 }
 
 data "template_file" "master_cloud_init_user_data" {
+  # needed when 0 master nodes are defined
+  count    = "${var.caasp_master_count}"
```
Comment (kiall) [inline caasp-kvm/cluster.tf]: Is 0 masters a valid option?

Comment (flavio) [inline caasp-kvm/cluster.tf]: I just wanted to provision the admin node to test this patch, plus other times in the past I tried to do a "hybrid" bootstrap of the cluster meaning: I created the admin node via dev-env and then I added all the nodes as regular vms started from the ISO.

File: caasp-kvm/tools/download_image.py [hunk 5]
```diff
@@ -133,3 +153,7 @@ def use_channel_file(url, force_redownload):
         use_local_file(args.url)
     if urlparse.urlparse(args.url).scheme == "channel":
         use_channel_file(args.url, args.force_redownload)
```
Comment (MaximilianMeister) [inline caasp-kvm/tools/download_image.py]: this line needs to be dropped

File: caasp-kvm/tools/download_image.py (hunk 3 above)
Comment (flavio) [inline caasp-kvm/tools/download_image.py]: Yes, sorry, I started working on that before the url got fixed on master. I'll fix it.

**Comment by kiall** [comment]
download image probably needs an update, I've ended up with:

    lrwxrwxrwx 1 kiall users   64 Aug  4 09:58 devel -> SUSE-CaaS-Platform-1.0-KVM-and-Xen.x86_64-1.0.0-Build30.38.qcow2
    lrwxrwxrwx 1 kiall users   44 Aug  4 10:16 head -> sles12-velum-development.x86_64-1.0.3.tar.xz

If the same channel name is used for both, this won't be very happy :)

**Comment by grahamhayes** [comment]
Doesn't seem to be working for me - when I run `./caasp-devenv -b -m 1 -w 3 -B` I am getting: 

```
>>> Wait until all minions are pending to be accepted
F

Failures:

  1) Boostrap cluster User accepts all minions
     Failure/Error: expect(page).to have_selector("a", text: "Accept Node", count: node_number, wait: 120)
       expected to find css "a" with text "Accept Node" 4 times but there were no matches. Also found "Logout", "Back", "", which matched the selector but not all filters.
     # ./spec/features/bootstrap_cluster.rb:39:in `block (3 levels) in <top (required)>'
     # ./spec/support/helpers.rb:8:in `with_screenshot'
     # ./spec/features/bootstrap_cluster.rb:38:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:93:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:93:in `block (2 levels) in <top (required)>'

Finished in 2 minutes 10.9 seconds (files took 0.52237 seconds to load)
3 examples, 1 failure

Failed examples:

rspec ./spec/features/bootstrap_cluster.rb:34 # Boostrap cluster User accepts all minions
```

But, the nodes are waiting to be accepted by salt:

```
caasp-admin:~ # docker exec -it $(docker ps | grep salt-master | awk '{print $1}') salt-key --list all 
Accepted Keys:
admin
ca
Denied Keys:
Unaccepted Keys:
4fbd504050534914bd7224c7467e872b
ab2ff9b6154949d9ae1041c0261094c5
e2ed6a45722e4882ad13d573f1a114f1
e931e3e83f834f438ea8b82faabc6a60
Rejected Keys:
```
Velums UI is not showing the nodes (its not just a test failure)
screenshot from 2017-08-09 12-54-15

**Comment by kiall** [comment]
@grahamhayes yea, salt is using the velum_production DB, while velum is using velum_development... So - no events get seen.

**Comment by MaximilianMeister** [comment]
> yea, salt is using the velum_production DB, while velum is using velum_development... So - no events get seen.

good point, haven't noticed that as i didn't run the bootstrapping

**Comment by flavio** [comment]
Latest commit fixes the issue pointed out by @grahamhayes. However this depends on https://github.com/kubic-project/velum/pull/283 to be merged into master

**Comment by grahamhayes** [comment]
The dashboard now fails to start - running the checkdb command gets:

```
caasp-admin:~ # docker exec -it 6e02e3410ae6 entrypoint.sh bundle exec rails runner /srv/velum/bin/check_db.rb
DB_DOWN
```

**Comment by grahamhayes** [comment]
Does https://github.com/kubic-project/caasp-container-manifests/blob/master/setup/mysql/setup-mysql.sh or https://github.com/kubic-project/caasp-container-manifests/blob/98e6f50e11e492cf324b47917264c0f47460eae8/public.yaml#L23-L46 need to be updated to ensure that the right DB is created and the velum user has access ?

**Comment by flavio** [comment]
I cleaned up the history of this PR. It now has all the relevant commits in a properly separated fashion.

**Comment by kiall** [comment]
Rebased, to pull in some required changes for CI (SSH Conf Generation)

**Pull request merged**